### PR TITLE
Feature/fa dataset schema sorting

### LINKED ISSUE #1690: FA compute wrong statistics under some conditions
Statistics computed by FA under some circumstances are obviously not the good numbers. 

Values computed for another colums seem to be used, which may come from issues with caching and/or with computing analytics for only a subset of the columns.

Image

Cache content after several requests:
```
de1324e1f76f433dae733755140ef16c
{'_computable_stats_cache': ['count', 'mean', 'std', 'sum', 'variance'],
 '_data': {'NODE_4558a41c-9afa-463c-9caf-3d7b8d83f0f7': {'AGE': {'count': 362,
                                                                 'mean': 0.7817679558011045,
                                                                 'variance': 0.17107941},
                                                         'PTEDUCAT': {'count': 362,
                                                                      'mean': 447.8729281767957,
                                                                      'variance': 69919.57}},
           'NODE_6abbf7a3-4460-46de-9980-fc74e15ab0fe': {'AGE': {'count': 538,
                                                                 'mean': 0.8345724907063194,
                                                                 'variance': 0.13831834},
                                                         'PTEDUCAT': {'count': 538,
                                                                      'mean': 450.59479553903367,
                                                                      'variance': 66121.71}}}}
602f22592686b2c9d21df06b44d04c23
{'_computable_stats_cache': ['count', 'mean', 'std', 'sum', 'variance'],
 '_data': {'NODE_4558a41c-9afa-463c-9caf-3d7b8d83f0f7': {'AGE': {'count': 362,
                                                                 'mean': 447.8729281767957,
                                                                 'variance': 69919.57}},
           'NODE_6abbf7a3-4460-46de-9980-fc74e15ab0fe': {'AGE': {'count': 538,
                                                                 'mean': 450.59479553903367,
                                                                 'variance': 66121.71}}}}
```

### PR DESCRIPTION
**Pull request by leonard2103**
Corrections to ensure ordered schemas and columns in lists. It prevents schemas ordered in different ways to make a new request and protects in node to ensure columns correspond to the order given by the dataset.

closes #1690 

**Developer Certificate Of Origin (DCO)**

By opening this merge request, you agree to the
[Developer Certificate of Origin (DCO)](https://github.com/fedbiomed/fedbiomed/blob/master/CONTRIBUTING.md#fed-biomed-developer-certificate-of-origin-dco)

This DCO essentially means that:

- you offer the changes under the same license agreement as the project, and
- you have the right to do that,
- you did not steal somebody else’s work.

**License**

Project code files should begin with these comment lines to help trace their origin:
```
# This file is originally part of Fed-BioMed
# SPDX-License-Identifier: Apache-2.0
```

Code files can be reused from another project with a compatible non-contaminating license.
They shall retain the original license and copyright mentions.
The `CREDIT.md` file and `credit/` directory shall be completed and updated accordingly.

**Guidelines for PR review**

General:

* give a glance to [DoD](https://fedbiomed.org/latest/developer/definition-of-done/)
* check [coding rules and coding style](https://fedbiomed.org/latest/developer/usage_and_tools/#coding-style)
* check docstrings (eg run `tests/docstrings/check_docstrings`)

Specific to some cases:

* update all conda envs consistently (`development` and `vpn`, Linux and MacOS)
* if modified researcher (eg new attributes in classes) check if breakpoint needs update (`breakpoint`/`load_breakpoint` in `Experiment()`, `save_state_breakpoint`/`load_state_breakpoint` in aggregators, strategies, secagg, etc.)
* if modified a component with versioning (config files, breakpoint, messaging protocol) then update the version following the rules in `common/utils/_versions.py`

### CODE REVIEW

File: fedbiomed/common/analytics/accumulators/_row.py [hunk 1]
```diff
@@ -92,7 +92,9 @@ def __init__(self, config: Dict[str, Any]):
                         self.vectorized_accumulators_classes[stat]()
                     )
 
-        logger.info(f"RowAccumulator initialized with columns: {self.column_order}")
+        logger.info(
+            f"RowAccumulator initialized with schema_columns: {self.schema_columns}"
+        )
```
Comment (srcansiz) [inline fedbiomed/common/analytics/accumulators/_row.py]: Hi @leonard2103, 

I think this log can be debug log rather then info.

File: tests/test_analytics/test_federated_analytics.py [hunk 2]
```diff
@@ -795,6 +795,71 @@ def test_fetch_stats_string_schema_normalized(self, mock_fa_job_cls, base_fa):
         assert call_kwargs["dataset_schema"] == ["age"]
 
 
+# ---------------------------------------------------------------------------
```
Comment (srcansiz) [inline tests/test_analytics/test_federated_analytics.py:798]: Hi @leonard2103,

It could be nice add a test for the case where there is empty string as column name to produce and the test the bug that is encountered.

File: fedbiomed/common/analytics/accumulators/_row.py [hunk 3]
```diff
@@ -37,19 +37,19 @@ def __init__(self, config: Dict[str, Any]):
         self.vectorized_output_map: Dict[str, Dict[int, int]] = {}
 
         self.column_configs = config.get("conf", {})
-        self.column_order = config.get("columns")
+        self.schema_columns = config.get("schema_columns")
```
Comment (srcansiz) [inline fedbiomed/common/analytics/accumulators/_row.py:40]: Maybe a stupid question but what is data does not have any column name ?

Comment (srcansiz) [reply fedbiomed/common/analytics/accumulators/_row.py:40]: I tested and it seems that column names are generated automatically while adding the dataset.

File: tests/test_analytics/test_federated_analytics.py (hunk 2 above)
Comment (leonard2103) [reply tests/test_analytics/test_federated_analytics.py:798]: It is treated as another column if exists. The error you encountered actually was happening with any column, the order was always the same. That is the reason to pass and use the schema of columns given by the dataset.

Comment (srcansiz) [reply tests/test_analytics/test_federated_analytics.py:798]: Ok I see, thanks for the explanation.

**Comment by mvesin** [comment]
@leonard2103 please don't forget to assign you issues when working on them :-)
